astropy/io/fits/header.py

```diff
@@ -1694,7 +1694,7 @@ def _keyword_from_index(self, idx):
         """
 
         if idx < 0:
-            idx += len(self._cards) - 1
+            idx += len(self._cards)
```

Comment:
Yep this was already identified in the original issue, but sadly there is no easy way to give credit (in git history) in this case.